ipatests/test_integration/test_commands.py

```diff
@@ -1489,3 +1489,28 @@ def test_ipa_cacert_manage_prune(self):
         self.master.run_command(['date', '-s', '-15Years'])
 
         assert isrgrootx1_nick in result
+
+    def test_client_doesnot_throw_responsenotready_error(self):
+        """
+        This testcase checks that ipa command
+        doesn't throw http.client.ResponseNotReady error
+        when current users session is deleted from the cache
+        """
+        error_msg = (
+            "http.client.ResponseNotReady: Request-sent \n"
+            "ipa: ERROR: an internal error has occurred"
+        )
+        domainname = "ipa.test"
+        del_ccache = "rm -fv /run/ipa/ccaches/admin@{}".format(domainname)
+
+        # kinit as admin on ipa-client and run ipa command
+        tasks.kinit_admin(self.clients[0], raiseonerr=False)
+        self.clients[0].run_command(["ipa", "user-find"])
+
+        # Delete the current user session cache on IPA server
+        cmd = self.master.run_command([del_ccache])
+
+        # Run the command on the client again after cache is removed
+        cmd = self.clients[0].run_command(["ipa", "user-find"])
+        assert cmd.returncode == 0
```

Comment:
No need to add this, `raiseonerr=True` is set by default.

Reply:
I agree with Rizwan, by default raiseonerr=True and the test will fail before reaching the `assert cmd.returncode == 0` line.
You have 2 choices:
- if you want to use the default raiseonerr=True, just call ipa user-show and delete the assertions on line 1529, 1520 + delete the error_msg on line 1514 as it won't be used. If ipa user-show fails, the test will exit on failure.
- if you want to check the output, add raiseonerr = True. This way you can check that error_msg is not in the output but in this case I would invert the 2 assertions.
IMO no need to look for a specific error message, you can stick to the default raiseonerr=True behavior.